File: collect_app/src/androidTest/java/org/odk/collect/android/support/CallbackCountingTaskExecutorRule.java
package org.odk.collect.android.support;

import androidx.arch.core.executor.testing.CountingTaskExecutorRule;

public class CallbackCountingTaskExecutorRule extends CountingTaskExecutorRule {

    private Runnable finishedCallback;

    public void setFinishedCallback(Runnable callback) {
        this.finishedCallback = callback;
    }

    @Override
    protected void onIdle() {
        if (finishedCallback != null) {
            finishedCallback.run();
        }
    }
}

File: collect_app/src/androidTest/java/org/odk/collect/android/support/rules/CallbackCountingTaskExecutorRule.kt
package org.odk.collect.android.support.rules

import androidx.arch.core.executor.testing.CountingTaskExecutorRule

class CallbackCountingTaskExecutorRule : CountingTaskExecutorRule() {

    private var finishedCallback: Runnable? = null

    fun setFinishedCallback(callback: Runnable?) {
        finishedCallback = callback
    }

    override fun onIdle() {
        finishedCallback?.run()
    }
}
